dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/JNDITest.java

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.websphere.simplicity.config.JNDIEntry;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+import jndiApp.JNDIservlet;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class JNDITest {
+
+    public static final String APP_NAME = "jndiApp";
+
+    @Server("jndiServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void copyAppToDropins() throws Exception {
+        server.deleteAllDropinApplications();
+        WebArchive webappWar = ShrinkWrap.create(WebArchive.class, "JNDIApplication.war").addClass(JNDIservlet.class);
+        ShrinkHelper.exportAppToServer(server, webappWar, DeployOptions.OVERWRITE);
+    }
+
+    @Test
+    public void testJNDIlookupNoEntry() throws Exception {
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS);
+        server.startServer();
+
+        ServerConfiguration config = server.getServerConfiguration();
+        JNDIEntry defaultEntry = new JNDIEntry();
+        defaultEntry.setJndiName("jndi/value");
+        defaultEntry.setValue("default value");
+        config.getJndiEntryElements().add(defaultEntry);
+        server.updateServerConfiguration(config);
+
+        server.stopServer();
```

Comment:
You should stop the server before doing the config update.  It probably doesn't effect the actual test execution because you take the checkpoint and restore the server process before doing the config update.  But when you have a server running while you do a config update it will dynamically react to the config change on running (restored) server.

We backup the workarea on checkpoint so when you restore the server again in the test it will start from the state when you did the checkpoint (before the config change) and then reprocess the config change.  That is technically still testing what we need it to, but it may confuse others looking at this test and wondering why the config is changed while the server is running.